vue/components/ui/molecules/carousel/carousel.vue

```diff
@@ -0,0 +1,283 @@
+    scale: 1.4;
+}
+
+.carousel .arrow-previous {
+    left: 0px;
+}
+
+.carousel .arrow-next {
+    right: 0px;
+}
+
+.carousel .arrow-previous:active {
+    transform: translate(-5px);
+}
+
+.carousel .arrow-next:active {
+    transform: translate(5px);
+}
+
+.carousel .slide-container {
+    height: 100%;
+    position: relative;
+    text-align: left;
+    white-space: nowrap;
+    width: 100%;
+}
+
+.carousel .slide-container .slide {
+    cursor: pointer;
+    display: inline-block;
+    height: 100%;
+    position: absolute;
+    text-align: center;
+    vertical-align: top;
+    width: 100%;
+}
+
+.carousel .slide-container .slide .slide-content {
+    align-items: center;
+    display: flex;
+    height: 100%;
+    justify-content: center;
+    padding: 0px 5px 0px 5px;
+    width: 100%;
+}
+
+.carousel .slide-container .slide .slide-content .image {
+    height: 100%;
+    object-fit: contain;
+    pointer-events: none;
+    width: 100%;
+}
+
+.fade-enter-active,
+.fade-leave-active {
+    transition: opacity 0.5s;
+}
+
+.fade-enter,
+.fade-leave-to {
+    opacity: 0;
+}
+</style>
+
+<script>
+export const Carousel = {
+    name: "carousel",
+    props: {
+        /**
+         * Set of images to be displayed in the carousel slider.
+         */
+        images: {
+            type: Array,
+            default: () => []
+        },
+        /**
+         * The index of the image to be displayed first.
+         */
+        imageIndex: {
+            type: Number,
+            default: 0
+        },
+        /**
+         * The width of the carousel in pixels.
+         */
+        width: {
+            type: Number,
+            default: 320
+        },
+        /**
+         * The height of the carousel in pixels.
+         */
+        height: {
+            type: Number,
+            default: 400
+        },
+        /**
+         * Weather or not the arrows to navigate between slides should show.
+         */
+        arrows: {
+            type: Boolean,
+            default: true
+        },
+        /**
+         * The color of the arrow to move between the slide images.
+         */
+        arrowsColor: {
+            type: String,
+            default: "#808080"
+        }
+    },
+    data: function() {
+        return {
+            dragStartPosition: null,
+            loadedImages: 0,
+            imageIndexData: this.imageIndex
+        };
+    },
+    computed: {
+        style() {
+            const base = {};
+            if (this.width) base.width = this.width + "px";
+            if (this.height) base.height = this.height + "px";
+            return base;
+        },
+        arrowsVisibility() {
+            return this.images.length > 1 && this.arrows;
+        }
+    },
+    watch: {
+        imageIndexData(value) {
+            this.$emit("update:image-index", value);
+        },
+        imageIndex(value) {
+            this.imageIndexData = value;
+        },
+        loadedImages(value) {

```

Comment:
What use-case are you seeing for this?

Reply:
This can be useful in a scenario where we need to load a lot of images. With this we could have a circular loading indicator that would update in sync with `update:load` to only show the carousel when all of the images were there.